### ISSUE TITLE
Missing script attributes after re-compiling with HMR

### ISSUE THREAD

**Issue by kumardeepakxyz** [webdiscus/pug-plugin#21]
Hi I'm facing two issues that are built on each other when using this plugin.

First, If I disable **Hot Module Replacement (HMP)** then page gets **infinite reload**.
Page keeps refreshing no matter what you do. However, enabling HMP fix this issue.

But this leads to another one.

Now with HMP enabled `defer` attribute added to `script` tag is randomly added/removed on liveReload.

Whenever I make any change to the code (especially SCSS), `defer` attribute added to `script(src=require(`~JS/${name}.js`) defer='defer')` is randomly added (majority of times it's not in the script) from the output.

**Comment by webdiscus** [comment]
Hi @kumardeepakxyz 

thank you for the bug report.
Yes, I can reproduce it.

If the attribute `defer` is already in Pug template, e.g.:
```pug
script(src=require('./main.js') defer='defer')
```

then:
- by first start `npm run start`, the attribute is rendered
- by any change after next re-compiling the attribute is missed

I will try to fix this very very strange issue.

**Comment by webdiscus** [comment]
@kumardeepakxyz 

In the version `2.7.5` is fixed issue with script attributes.

>  If I disable Hot Module Replacement (HMP) then page gets infinite reload

I can't reproduce this issue. Can you please share the  webpack config?

Here is working configuration for dev server with HMR:

```js
devServer: {
  static: {
    directory: path.join(__dirname, 'public'),
  },

  watchFiles: {
    paths: ['src/**/*.*'],
    options: {
      usePolling: true,
    },
  },
},
```

**Comment by kumardeepakxyz** [comment]
@webdiscus that is great!

It has fixed the issue with the `defer` attribute.

But the `infinite reload loop` is still there for me - now with both HMP Enabled/Disabled.

When I start the server, it is fine at first - as soon as I make any change it keeps reloading the page.

But strangely I noticed that if I open the terminal it stops reloading and gives the following HMP warning(s) in the console.

![Screenshot 2022-06-19 at 8 48 50 AM](https://user-images.githubusercontent.com/36254505/174464593-3af054bb-a143-49f9-96e8-6a8bb38e9809.png)

Also I have added video below to see it live:

https://user-images.githubusercontent.com/36254505/174464638-7d7fc767-c3c3-42b5-a839-60b3bdd4eef0.mp4

My **devServer** config is as follow:

```
devServer: {
    devMiddleware: {
      publicPath: join(__dirname, 'build'),
      writeToDisk: true,
    },
    port: '9999',
    proxy: {
      '/': {
        target: `http://localhost/${projectName}/build`,
      },
    },
    static: {
      directory: join(__dirname, 'build'),
    },
    watchFiles: {
      paths: ['src/**/*.*'],
      options: {
        usePolling: true,
      },
    },
  },
```

Hope all this give you some direction in order to help this issue.
